src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -24,96 +37,145 @@
  *
  * See https://github.com/CrossRef/rest-api-doc
  */
-public class CrossRef {
-    private static final Log LOGGER = LogFactory.getLog(CrossRef.class);
+public class CrossRef implements IdParserFetcher<DOI>, EntryBasedParserFetcher, SearchBasedParserFetcher, IdBasedParserFetcher {
```

Comment:
My opinion on this:  Don't combine all interfaces to a single one. The purpose of an Interface is to provide specific methods/behaviour. 
If one concrete fetcher has multiple behavior, e.g. can fetch IDs as well as FullText, it should implement both interfaces.